src/Microsoft.AspNetCore.Blazor.Razor.Extensions/ComplexAttributeContentPass.cs

```diff
@@ -73,17 +74,59 @@ private static bool TrySimplifyContent(IntermediateNode node)
                 //
                 // There's a special case here for explicit expressions. See https://github.com/aspnet/Razor/issues/2203
                 // handling this case as a tactical matter since it's important for lambdas.
-                if (cSharpNode.Children.Count == 3 &&
-                    cSharpNode.Children[0] is IntermediateToken token0 &&
-                    cSharpNode.Children[2] is IntermediateToken token2 &&
-                    token0.Content == "(" &&
-                    token2.Content == ")")
                 {
-                    cSharpNode.Children.RemoveAt(2);
-                    cSharpNode.Children.RemoveAt(0);
+                    if (cSharpNode.Children.Count == 3 &&
+                        cSharpNode.Children[0] is IntermediateToken token0 &&
+                        cSharpNode.Children[2] is IntermediateToken token2 &&
+                        token0.Content == "(" &&
+                        token2.Content == ")")
+                    {
+                        cSharpNode.Children.RemoveAt(2);
+                        cSharpNode.Children.RemoveAt(0);
+
+                        // We were able to simplify it, all good.
+                        return false;
+                    }
+                }
+
+                // There's yet another special case when you have an attribute with a template
+                // like <div attr="@(@<div>Hi</div>)"
```

Comment:
So the story with templates inside attributes. I thought it woulnd't work, but it does. Well mostly. Colorization doesn't work in VS. Completion of C# inside the attribute works.  Basically with a bunch of workarounds we can have templates work inside component attributes. 

This seems worthwhile enough to me to tolerate a bit of tooling pain for now. Thoughts?

Reply:
Let's discuss in the sync today.